src/Pact/Types/Advice.hs

```diff
@@ -72,33 +73,33 @@ instance Show (AdviceContext r) where
 
 -- | Bracket some Pact operation.
 newtype Advice = Advice {
-  -- | Callback for advising a pact operation. '_advise i ctx op'
-  -- gives source info in 'i', advice context in 'ctx', and 'op'
-  -- as the bracketed action. The implementation is expected to
-  -- execute 'op' whose return value '(r,a)' contains the ultimate
-  -- return value in 'a', with 'r' available for inspection.
-  _advise :: forall m r a . MonadIO m
+  -- | Callback for advising some pact operation. '_advise i ctx'
+  -- is invoked before the operation with location info in 'i' and
+  -- advice context in 'ctx', and returns a continuation for
+  -- processing the operation return type post-execution.
+  _advise :: forall m r . MonadIO m
     => Info
     -> AdviceContext r
-    -> m (r, a)
-    -> m a
+    -> m (r -> m ())
```

Comment:
is there a reason this is in `m (r -> m ())`? Like is there a reason to potentially want to side effect before we even use the function?

How about `Info -> AdviceContext r -> r -> m ()`

Reply:
what's the difference of invoking it pre-op vs post-op when the actual function that side effects is invoked always post-op? Or am I missing something here?